### PR TITLE
Enhancement: Add support for rector/rector:~0.6.0

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] adds support for `rector/rector:~0.6.0`

Fixes #167.

### CODE REVIEW

File: generated/misc.php [hunk 1]
```diff
@@ -133,6 +133,27 @@ function sapi_windows_cp_set(int $cp): void
...
+ *
+ * @param int $event The CTRL even to send;
+ * either PHP_WINDOWS_EVENT_CTRL_C
+ * or PHP_WINDOWS_EVENT_CTRL_BREAK.
+ * @param int $pid The ID of the process to which to send the event to. If 0
+ * is given, the event is sent to all processes of the process group.
+ * @throws MiscException
+ *
+ */
+function sapi_windows_generate_ctrl_event(int $event, int $pid = 0): void
```
Comment (localheinz) [inline generated/misc.php]: 🤔

**Comment by Kharhamel** [comment]
conflicts again. please remove the commit where you regenerate the file.

**Comment by localheinz** [comment]
@Kharhamel 

Rebased!

File: rector-migrate.yml [hunk 2]
```diff
@@ -1,1113 +1,1114 @@
...
-    yaz_database: 'Safe\yaz_database'
-    yaz_element: 'Safe\yaz_element'
-    yaz_present: 'Safe\yaz_present'
-    yaz_search: 'Safe\yaz_search'
-    yaz_wait: 'Safe\yaz_wait'
-    zip_entry_close: 'Safe\zip_entry_close'
-    zip_entry_open: 'Safe\zip_entry_open'
-    zip_entry_read: 'Safe\zip_entry_read'
-    zlib_decode: 'Safe\zlib_decode'
+    $oldFunctionToNewFunction
```
Comment (Kharhamel) [inline rector-migrate.yml]: isn't it a typo?

Comment (Kharhamel) [reply rector-migrate.yml]: It looks like line is making the pipeline fail

File: .travis.yml [hunk 3]
```diff
@@ -61,7 +61,9 @@ script:
     echo "Generated files are different from commited files. Please run './safe.php generate' command and commit the results."
     exit 1;
   fi
+- cd generator/tests/rector/default && composer install && composer rector && composer test && cd ../../../..
```
Comment (localheinz) [inline .travis.yml]: This ensures that using `rector-migrate.yml` is still possible.

**Comment by localheinz** [comment]
@Kharhamel 

This works fine now, do you want to take another look?

**Comment by Kharhamel** [comment]
I don't know much about rector. If I understand correctly, we allow rector 0.6 support while still keeping the compatibility with 0.5? If so, i will merge it.

**Comment by localheinz** [comment]
@Kharhamel 

Yes, that's it.

Since `1.0.0` has been tagged, the previous change would be breaking. 

This way, users of `rector/rector:~0.5.0` and `rector/rector:~0.6.0` can both use the YAML configuration files.

**Comment by Aerendir** [comment]
If of any help, this works for me using Rector v0.6.

**Comment by localheinz** [comment]
Perhaps @TomasVotruba can have a quick look and confirm?

**Comment by TomasVotruba** [comment]
Better go with 0.6.7. There've been some changes

File: README.md [hunk 4]
```diff
@@ -111,18 +111,42 @@ find these functions but changing the namespace of the functions one function at
...
-First, you need to install Rector:
+### `rector/rector:~0.5.0`
+
+Run
+
+```bash
+$ composer require --dev rector/rector:~0.5
+```
+
+to install `rector/rector:~0.5`.
```
Comment (TomasVotruba) [inline README.md]: This part can be dropped, as very obsolete version. It also conflicts with PHPStan 0.11-, so extra troubles for no gain

File: README.md [hunk 5]
```diff
@@ -111,18 +111,42 @@ find these functions but changing the namespace of the functions one function at
...
 ```
 
-Now, you simply need to run Rector with this command:
+to install `rector/rector:~0.6`.
+
+Run
 
 ```bash
-vendor/bin/rector process src/ --config vendor/thecodingmachine/safe/rector-migrate.yml
+$ vendor/bin/rector process src/ --config vendor/thecodingmachine/safe/rector-migrate-0.6.yml
```
Comment (TomasVotruba) [inline README.md]: Better support just one minor Rector version, this is rather confusing even to me

File: generator/tests/rector/0.6/composer.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{
+  "autoload": {
+    "psr-4": {
+      "Test\\": "src/"
+    }
+  },
+  "require": {
+    "php": ">=7.1",
+    "rector/rector": "^0.6"
```
Comment (TomasVotruba) [inline generator/tests/rector/0.6/composer.json]: Rector 0.6 require PHP 7.2, so it can be bumped as well to be real

Comment (Aerendir) [reply generator/tests/rector/0.6/composer.json]: Updating the minimum PHP version is a BC. This requires a new major version of the entire library...

**Comment by Aerendir** [comment]
I'm an happy user of both Rector and Safe.

Reading your comments, one strange idea came to my mind: what if the support for Safe is provided directly by Rector?

This way, Safe remain an independent very useful package that doesn't break each time Rector is updated...

As told, this is a strange idea 😅

File: generator/tests/rector/0.6/composer.json (hunk 6 above)
Comment (TomasVotruba) [reply generator/tests/rector/0.6/composer.json]: This a test file.

And bumping PHP version [is not a BC break](https://www.doctrine-project.org/2017/07/25/php-7.1-requirement-and-composer.html), as it only narrows scope.

**Comment by TomasVotruba** [comment]
> Reading your comments, one strange idea came to my mind: what if the support for Safe is provided directly by Rector?

Well you're right, that's what 99 % projects do :D so feel free to send PR there!

File: README.md (hunk 4 above)
Comment (moufmouf) [reply README.md]: :+1:

My take on this: we can keep support for rector:0.5 (in order to not break compatibility with users using old Rector) but we should definitely stop advertising this possibility in the README (the README should only contain information about the latest minor version)

**Comment by localheinz** [comment]
Thank you for your feedback, @TomasVotruba - I made a few adjustments here!

**Comment by localheinz** [comment]
Thank you, @Aerendir, @Kharhamel, @moufmoufm and @TomasVotruba!

**Comment by Aerendir** [comment]
> > Reading your comments, one strange idea came to my mind: what if the support for Safe is provided directly by Rector?
> 
> Well you're right, that's what 99 % projects do :D so feel free to send PR there!
> 
> EDIT: Think of it as Symfony Flex with just with upgrade recipes

I don't understand... Where should have I to submit the PR? I'm not the owner nor I'm involved in the Safe library: I have no decision power about it...

What should have I to do?

**Comment by TomasVotruba** [comment]
The example would the best. If something in Symfony 3.3 changes, it should be added here:
https://github.com/rectorphp/rector/blob/master/config/set/symfony/symfony33.yaml

**Comment by Aerendir** [comment]
@TomasVotruba , ok, now it is clear: so, it sufficient to copy the current file `rector-migrate*.yml` from Safe to `https://github.com/rectorphp/rector/blob/master/config/set/thecodingmachine/safe10.yaml`?

@localheinz , WDYT about this? Can I submit the PR to Rector: have I your permission?

**Comment by TomasVotruba** [comment]
> @TomasVotruba , ok, now it is clear: so, it sufficient to copy the current file rector-migrate*.yml from Safe to https://github.com/rectorphp/rector/blob/master/config/set/thecodingmachine/safe10.yaml?

:+1:

**Comment by Aerendir** [comment]
Ping

**Comment by localheinz** [comment]
@Aerendir 

Perhaps we can find a way to automatically open a pull request whenever we have any changes in the corresponding YAML file?

Maybe using https://github.com/gr2m/create-or-update-pull-request-action?

**Comment by TomasVotruba** [comment]
There you go: https://github.com/rectorphp/rector/pull/3078

You can update the set path:

```diff
-vendor/bin/rector process src --config vendor/thecodingmachine/safe/rector-migrate-0.7.yml
+vendor/bin/rector process src --config vendor/rector/rector/config/set/thecodingmachine/safe07.yaml
```

**Pull request merged**